src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -435,4 +513,74 @@ export default class AircraftController {
             this.aircraft.list[i].fms.my_aircrafts_eid = i;
         }
     }
+
+    /**
+     * Used to build up the appropriate data needed to instantiate an `AircraftInstanceModel`
+     *
+     * @for AircraftController
+     * @method _buildAircraftProps
+     * @param spawnModel {SpawnPatternModel}
+     * @return {object}
+     * @private
+     */
+    _buildAircraftProps(spawnModel) {
+        const airlineId = spawnModel.getRandomAirlineForSpawn();
+        const { name, fleet } = airlineNameAndFleetHelper([airlineId]);
+        const airlineModel = this.airlineController.findAirlineById(name);
+        const aircraftDefinition = this._getRandomAircraftTypeDefinitionForAirlineId(airlineId, airlineModel);
+        const destination = this._setDestinationFromRouteOrProcedure(spawnModel);
```

Comment:
I've figured out the intent of this method (you tried to tell me but I was too dumb to listen). `AircraftInstanceModel.destination` is used (stupidly) to determine where it's going; for arrivals, that's the airport, and for departures, that has been the departure procedure (*barfs*).

Part of this refactor should include the addition of `AircraftInstanceModel.origin`, which is used similarly to `AircraftInstanceModel.destination`, and can ONLY be an airport. Alternatively, this information could also be placed within `AircraftInstanceModel.AicraftFlightManagementSystem.fp`.

There will be one or two places where `AircraftInstanceModel.destination` is used to reference the procedure the aircraft is on, but most of them will have been wiped out by the work on `spawnPatternModel`.

This would be a step in the right direction with regards to eventually doing away with an aircraft being of a given category. They shouldn't be looked at so strictly as "this is an arrival; this is a departure." If we have a large center sector, we could often have VFR aircraft who want to fly from tiny airport A to tiny airport B, both of which are within our airspace. Then they would be a "departure", "arrival", and "overflight" at the same time. </aimless_yammering>